feature/amazon/src/test/java/com/revenuecat/purchases/amazon/AmazonBackendTest.kt

```diff
@@ -0,0 +1,136 @@
+package com.revenuecat.purchases.amazon
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.PurchasesErrorCode
+import com.revenuecat.purchases.amazon.helpers.successfulRVSResponse
+import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.HTTPClient
+import com.revenuecat.purchases.utils.SyncDispatcher
+import io.mockk.every
+import io.mockk.mockk
+import org.assertj.core.api.Assertions
+import org.assertj.core.api.Assertions.assertThat
+import org.json.JSONObject
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.io.IOException
+
+private const val API_KEY = "TEST_API_KEY"
+
+@RunWith(AndroidJUnit4::class)
+class AmazonBackendTest {
+
+    private var mockClient = mockk<HTTPClient>()
+
+    private var backend: Backend = Backend(
+        API_KEY,
+        SyncDispatcher(),
+        mockClient
+    )
+
+    private var receivedOnSuccess: JSONObject? = null
+    private var receivedError: PurchasesError? = null
+
+    private val expectedOnSuccess: (JSONObject) -> Unit = {
+        receivedOnSuccess = it
+    }
+
+    private val unexpectedOnSuccess: (JSONObject) -> Unit = {
+        Assertions.fail("Shouldn't be success.")
+    }
+
+    private val expectedOnError: (PurchasesError) -> Unit = {
+        receivedError = it
+    }
+
+    private val unexpectedOnError: (PurchasesError) -> Unit = { _ ->
+        Assertions.fail("Shouldn't be error.")
+    }
+
+    private var underTest = AmazonBackend(backend)
+
+    private var successfulResult = HTTPClient.Result(
+        responseCode = 200,
+        body = successfulRVSResponse()
+    )
+    private var unsuccessfulResult = HTTPClient.Result(
+        responseCode = 401,
+        body = JSONObject(
+            """
+                {
+                    "code":7225,"message":
+                    "Invalid API Key."
+                }
+            """.trimIndent()
+        )
+    )
+
+    @Before
+    fun setup() {
+        receivedOnSuccess = null
+    }
+
+    @Test
+    fun `successfully getting Amazon receipt data`() {
+        every {
+            mockClient.performRequest(
+                path = "/receipts/amazon/store_user_id/receipt_id",
+                body = null,
+                headers = mapOf("Authorization" to "Bearer $API_KEY")
+            )
+        } returns successfulResult
+
+        underTest.getAmazonReceiptData(
+            receiptId = "receipt_id",
+            storeUserID = "store_user_id",
+            expectedOnSuccess,
+            unexpectedOnError
+        )
+
+        assertThat(receivedOnSuccess!!).isNotNull
```

Comment:
it will always be null, note how I am passing `unexpectedOnError`, which will print a nice error message in the case the error is called:

```
    private val unexpectedOnError: (PurchasesError) -> Unit = { _ ->
        Assertions.fail("Shouldn't be error.")
    }
    ```